packages/store/src/patterns/patternMatchers.js

```diff
@@ -354,17 +433,42 @@ const makePatternKit = () => {
    * @returns {boolean}
    */
   const checkMatchesInternal = (specimen, patt, check) => {
-    if (isKey(patt)) {
-      // Takes care of all patterns which are keys, so the rest of this
-      // logic can assume patterns that are not key.
-      return check(keyEQ(specimen, patt), X`${specimen} - Must be: ${patt}`);
-    }
-    assertPattern(patt);
-    const specimenStyle = passStyleOf(specimen);
-    const pattStyle = passStyleOf(patt);
-    switch (pattStyle) {
+    // Worth being a bit verbose and repetitive in order to optimize
+    const patternKind = kindOf(patt, check);
+    const specimenKind = kindOf(specimen); // may be undefined
+    switch (patternKind) {
+      case undefined: {
+        return assert.fail(X`pattern expected: ${patt}`);
+      }
+      case 'promise': {
+        return assert.fail(X`promises cannot be patterns: ${patt}`);
+      }
+      case 'error': {
+        return assert.fail(X`errors cannot be patterns: ${patt}`);
+      }
+      case 'undefined':
+      case 'null':
+      case 'boolean':
+      case 'number':
+      case 'bigint':
+      case 'string':
+      case 'symbol':
+      case 'copySet':
+      case 'copyBag':
+      case 'remotable': {
+        // These kinds are necessarily keys
+        return check(keyEQ(specimen, patt), X`${specimen} - Must be: ${patt}`);
```

Comment:
since it's all the leaeves, put this assertion in an `if` statement?

Reply:
Excellent suggestion. Resulted in a small but consistently measurable speedup. Doubled the `empty` ops/sec! (Which is exactly the case we'd expect is most sensitive to this.)

Done.